File: ruby/fizzbuzz/fizzbuzz.rb
# FizzBuzz - Write a function to print out the numbers 1 to 100.
# Except numbers divisible by 3, print "FIZZ", numbers divisible by 5, print "BUZZ", numbers divisible by 3 and 5, print "FIZZBUZZ"

def fizzbuzz(number)
  index = 1
  while index < number
    if index % 3 == 0 && index % 5 == 0
      p "FIZZBUZZ"
    elsif index % 3 == 0
      p "FIZZ"
    elsif index % 5 == 0
      p "BUZZ"
    else
      p index
    end
    index += 1
  end
end

fizzbuzz(100)

File: javascript/Number/fizzbuzz/fizzbuzz.js
// FizzBuzz - Write a function to print out the numbers 1 to 100.
// Except numbers divisible by 3, print "FIZZ", numbers divisible by 5, print "BUZZ", numbers divisible by 3 and 5, print "FIZZBUZZ"

const fizzbuzz = (number) => {
  let index = 1
  while (index < number) {
    if (index % 3 === 0 && index % 5 === 0) {
      console.log("FIZZBUZZ")
    } else if (index % 3 === 0) {
      console.log("FIZZ")
    } else if (index % 5 === 0) {
      console.log("BUZZ")
    } else {
      console.log(index)
    }
    index += 1
  }
}
fizzbuzz(100)